Servers/Administration/OJS.Servers.Administration/Controllers/ProblemsController.cs

```diff
@@ -430,29 +438,17 @@ await this.contestsValidationHelper
             IsReadOnly = true,
         });
 
-        if (!contest.IsOnlineExam)
+        if (contest.IsOnlineExam)
         {
-            formControls.Add(new FormControlViewModel
-            {
-                Name = AdditionalFormFields.ProblemGroupType.ToString(),
-                Options = EnumUtils.GetValuesFrom<ProblemGroupType>().Cast<object>(),
-                Type = typeof(ProblemGroupType),
-                Value = entity.ProblemGroup?.Type ?? default(ProblemGroupType),
-            });
+            var problemGroupFieldType = formControls.First(x => x.Name == AdditionalFormFields.ProblemGroupType.ToString());
 
+            problemGroupFieldType.IsHidden = true;
+        }
+        else
+        {
             var problemGroupField = formControls.First(x => x.Name == nameof(Data.Models.Problems.Problem.ProblemGroup));
 
-            if (action == EntityAction.Create)
-            {
-                // On Create, we should always create new ProblemGroup for the Problem,
-                // not allow attaching the Problem to an existing ProblemGroup
-                formControls.Remove(problemGroupField);
```

Comment:
Why we do not need to `Remove` anymore?

Reply:
As far as I have checked, Remove and  Hidden have the same logic. I removed it for simplicity.